custom_components/toyota/sensor.py

```diff
@@ -293,8 +305,9 @@ async def async_setup_entry(
                 vehicle_index=index,
                 description=description,
                 unit=unit,
+                suggested_unit=unit2,
             )
-            for capability, description, sensor_type, unit in capabilities_descriptions
+            for capability, description, sensor_type, unit, unit2 in capabilities_descriptions

```

Comment:
In my opinion, the general solution now looks much better 😊 
But I don't like the fact that we now have two very generic names here, `unit` and `unit2`, where it's not immediately clear what they are for.
I think we should also rename the parameters to `native_unit` and `suggested_unit` at this point, so that their purpose is immediately clear.

Reply:
Just add `# noqa: E501` at the end of the line for now:
```python
            for capability, description, sensor_type, unit, suggested_unit in capabilities_descriptions  # noqa: E501
            if capability
```

Reply:
I added `# noqa: E501` to the line, as well as renamed the units to `native_unit` and `suggested_unit`. Hopefully this should resolve the issue.